cucumber/api/features/login.feature

```diff
@@ -81,6 +81,10 @@ Feature: Exchange a role's password for its API key
       cucumber:user:non-exist failed to login with authenticator authn service cucumber:webservice:conjur/authn: CONJ00007E 'non-exist' not found
     """
 
+  Scenario: Empty username results in 401 response
```

Comment:
what about `/authenticate` requests? can we add a test in `authenticate.feature`?

Reply:
I would add uniq chars and space between two words as well

Reply:
Uniq chars? Eg single character user name? or a string of a single repeated character?

Reply:
something like` !@#$%^&*(){}` to make sure it doesnt fail as part of reading those chars

Reply:
Not sure the same test makes sense for authenticate as the route won't match with an empty string, so it will result in 404:

Correctly formed authenticate request (conjur-quickstart local docker instance):
```
root@4d8ed4895116:/# curl -i --header "Accept-Encoding: base64" --data "${key}" http://conjur/authn/conjur/admin/authenticateHTTP/1.1 200 OK
Content-Encoding: base64
Content-Type: text/plain; charset=utf-8
ETag: W/"76fc6a0836156143887cec2a33197620"
Cache-Control: max-age=0, private, must-revalidate
X-Request-Id: 7597cc83-a4e5-4553-b260-6a661b9b2d15
X-Runtime: 0.059081
Transfer-Encoding: chunked

eyJwcm... snip rest of token
```
Authenticate request with empty username:
```
root@4d8ed4895116:/# curl -i --header "Accept-Encoding: base64" --data "${key}" http://conjur/authn/conjur//authenticate
HTTP/1.1 404 Not Found
Content-Type: text/html; charset=utf-8
X-Request-Id: 25413780-9c20-4896-bc3e-1986b0cc6ca0
X-Runtime: 0.015727
```
In this case the error is 404, which is reasonable given the username is part of the route.

Reply:
you are correct, my bad